contents/testing/integration-testing.md

```diff
@@ -0,0 +1,214 @@
+
+:fas-info-circle: A more detailed guide on using specific methods for incremental testing together with examples can be found [here](https://www.softwaretestinghelp.com/incremental-testing/).
+
+:fas-info-circle: [NUS Software Engineer textbook](https://se-education.org/se-book/integration/#integration) also offer some concrete examples of different approaches of integration testing.
+
+### 2. Design the test scenarios, cases and scripts
+
+Before the actual coding is done, a basic test strategy deciding the test cases and test data used should be crafted. This usually involves setting a <span style="color:red">test case ID</span>, <span style="color:red">objective</span>, <span style="color:red">description</span> and <span style="color:red">expected result</span>.
+Using the example shown above, below shows a sample integration test used for the `login` and `ordering` modules:
+
+~~~
+Test case ID: 1
+Objective: Check the link between login and ordering modules
+Description: Enter login credentials and click on login button
+Expected result: To be directed to order food page based on the login user
+~~~
+
+<box type="warning">As integration test cases are <tooltip content="It generally takes longer time due to additional overheads such as waiting for dependent modules to respond">expensive operations</tooltip> compared to unit testing, it should focus mainly on the integration of modules together and not on specific actions within the same module.</box>
+
+:fas-info-circle: More details about the ways to structure incremental testing can be found on [this article](https://www.softwaretestinggenius.com/various-approaches-in-integration-testing/)
+
+### 3. Execute the test scripts and report defects if any
+
+Depending on the approach you have chosen for your integration plan and the test cases, the way you execute your code for testing will differ.
+
+- Big bang approach usually requires all the modules to be developed before you can start with the integration testing.
+
+- Incremental approach is usually conducted simultaneously with the module's development. Stubs and drivers are used to mimic the modules for writing tests since not all modules are developed at the point of integration testing.
+[NUS Software Engineer module site](https://nus-cs2103-ay1819s1.github.io/cs2103-website/se-book-adapted/chapters/testing.html#stubs) provides concrete codes for using stubs.
+
+:fas-exclamation-circle: Either way, such test scripts can help you in singling out specific integration errors to be resolved before it gets more serious. Hence, you should ensure that all high prioritized bugs are fixed before moving on. 
+
+:fas-info-circle: [This article](https://www.guru99.com/test-environment-software-testing.html) shows more details on how to set up a test environment for better integration testing.
```

Comment:
> This article

Do you think it would be better to change all similar hyperlinks to specify the resource and the title as well? Or do you think it's not required?